src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts

```diff
@@ -9,7 +12,48 @@ export const getExpenditureStep = (status?: ExpenditureStatus) => {
       return ExpenditureStep.Review;
     case ExpenditureStatus.Locked:
       return ExpenditureStep.Funding;
+    case ExpenditureStatus.Finalized:
+      return ExpenditureStep.Payment;
     default:
       return ExpenditureStep.Create;
   }
 };
+
+/**
+ * Returns a boolean indicating whether the expenditure is fully funded,
+ * i.e. the balance of each token is greater than or equal to the sum of its payouts
+ */
+export const isExpenditureFullyFunded = (expenditure?: Expenditure | null) => {
+  if (!expenditure) {
+    return false;
+  }
+
+  if (!expenditure.balances) {
+    return false;
+  }
+
+  const amountsByToken = expenditure.slots.map((slot) => {
+    const amounts = {};
+
+    slot.payouts?.forEach((payout) => {
+      if (!payout.isClaimed) {
+        amounts[payout.tokenAddress] = BigNumber.from(
+          amounts[payout.tokenAddress] ?? 0,
+        ).add(payout.amount);
+      }
+    });
+
+    return amounts;
+  });
+
+  return amountsByToken.every((token) => {
+    const [tokenAddress] = Object.keys(token);
```

Comment:
I am not sure I am a fan of this, could we construct the object differently so we don't have to take the first key and then checking the amount?